src/MatrixFields/field_matrix_iterative_solver.jl

```diff
@@ -0,0 +1,465 @@
+
+function field_matrix_solver_cache(alg::StationaryIterativeSolve, A, b)
+    P_cache = preconditioner_cache(alg.P_alg, A, b)
+    # Note: We cannot use similar_to_x here because it doesn't work for some
+    # particularly complicated unit tests. For now, we will assume that x is
+    # similar to b, rather than just keys(x) == keys(b).
+    previous_x_cache =
+        alg.correlated_solves ? (; previous_x = zero.(similar(b))) : (;)
+    return (; P_cache, previous_x_cache...)
+end
+
+check_field_matrix_solver(alg::StationaryIterativeSolve, cache, A, b) =
+    check_preconditioner(alg.P_alg, cache.P_cache, A, b)
+
+function run_field_matrix_solver!(alg::StationaryIterativeSolve, cache, x, A, b)
+    P = concrete_preconditioner(alg.P_alg, cache.P_cache, A)
+    if alg.print_radius
+        e₀ = concrete_field_vector(b) # Initialize e to any nonzero vector.
+        eigenvalues, _, info = eigsolve(e₀, 1; alg.eigsolve_kwargs...) do e
+            e_view = field_vector_view(e, keys(b).name_tree)
+            lazy_A_e = lazy_mul(A, e_view)
+            lazy_invP_A_e =
+                apply_preconditioner(alg.P_alg, cache.P_cache, P, lazy_A_e)
+            concrete_field_vector(@. e_view - lazy_invP_A_e)
+        end
+        if info.converged == 0
+            (; tol, maxiter) = alg.eigsolve_kwargs
+            @warn "Unable to approximate ρ(I - inv(P) * A) to within a \
+                   tolerance of $(100 * tol) % in $maxiter or fewer iterations"
+        else
+            spectral_radius = abs(eigenvalues[1])
+            if spectral_radius < 1
+                @info "ρ(I - inv(P) * A) ≈ $spectral_radius"
+            else
+                @warn "StationaryIterativeSolve may not converge because \
+                       ρ(I - inv(P) * A) ≈ $spectral_radius is not less than 1"
+            end
+        end
+    end
+    if alg.correlated_solves
+        @. x = cache.previous_x
+    else
+        @. x = zero(x)
+    end
+    for iter in 1:(alg.n_iters)
+        lazy_Δb = lazy_sub(b, lazy_mul(A, x))
+        lazy_Δx = apply_preconditioner(alg.P_alg, cache.P_cache, P, lazy_Δb)
+        if alg.print_norm
+            norm_Δx = norm(concrete_field_vector(Base.materialize(lazy_Δx)))
+            @info "||x[$(iter - 1)] - x'||₂ ≈ $norm_Δx"
+        end
+        @. x += lazy_Δx
+    end
+    if alg.print_norm
+        lazy_Δb = lazy_sub(b, lazy_mul(A, x))
+        lazy_Δx = apply_preconditioner(alg.P_alg, cache.P_cache, P, lazy_Δb)
+        norm_Δx = norm(concrete_field_vector(Base.materialize(lazy_Δx)))
+        @info "||x[$(alg.n_iters)] - x'||₂ ≈ $norm_Δx"
+    end
+    if alg.correlated_solves
+        @. cache.previous_x = x
+    end
+end
+
+"""
+    ApproximateBlockArrowheadIterativeSolve(names₁...; [P_alg₁], [alg₁], [alg₂], [kwargs...])
+
+Shorthand for constructing a `BlockLUDecompositionSolve` that wraps a
+`StationaryIterativeSolve` with a `BlockArrowheadSchurComplementPreconditioner`.
+
+TODO: Add some motivation and references.
```

Comment:
???

Reply:
Yeah, I'm currently trying to figure out how to document this clearly. This is also called a "Schur complement solve" in the literature, and it looks like this is a "standard" way to approximately solve advection-diffusion problems in CFD. [Here](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7365595/) is a paper that somewhat describes this, though I'm looking for a better one to cite in this docstring.